### PR TITLE
cdc: add description about the checksum v2

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18273
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Oreoxmt — APPROVED** [review]

File: ticdc/ticdc-integrity-check.md [hunk 1]
```diff
@@ -5,15 +5,7 @@ summary: Introduce the implementation principle and usage of the TiCDC data inte
...
-
-TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka Consumer reads data, it calculates a new checksum using the same algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka Consumer.
-
-For more information about the algorithm of the checksum, see [Algorithm for checksum calculation](#algorithm-for-checksum-calculation).
+Starting from v7.1.0, TiCDC introduces the data integrity validation feature, which uses a [checksum algorithm](#checksum-algorithm) to validate the integrity of single-row data. This feature helps verify whether any error occurs in the process of writing data from TiDB, replicating it through TiCDC, and then writing it to a Kafka cluster. The data integrity validation feature only supports changefeeds that use Kafka as the downstream and currently supports the simple and Avro protocol. For more information about the algorithm of the checksum, see [Algorithm for checksum calculation](#algorithm-for-checksum-calculation).
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:8]:
```suggestion
Starting from v7.1.0, TiCDC introduces the data integrity validation feature, which uses a [checksum algorithm](#checksum-algorithms) to validate the integrity of single-row data. This feature helps verify whether any error occurs in the process of writing data from TiDB, replicating it through TiCDC, and then writing it to a Kafka cluster. The data integrity validation feature only supports changefeeds that use Kafka as the downstream and currently supports the simple and Avro protocol. For more information about the algorithm of the checksum, see [Algorithm for checksum calculation](#algorithm-for-checksum-calculation).
```

File: ticdc/ticdc-integrity-check.md [hunk 2]
```diff
@@ -67,6 +59,22 @@ TiCDC disables data integrity validation by default. To disable this feature aft
...
+
+Before v8.4.0, TiDB and TiCDC use Checksum V1 for checksum calculation and verification.
+
+After you enable the checksum integrity validation feature for single-row data, TiDB uses the CRC32 algorithm to calculate the checksum of a row and writes it to TiKV along with the data. TiCDC reads the data from TiKV and recalculates the checksum using the same algorithm. If the two checksums are equal, it indicates that the data is consistent during the transmission from TiDB to TiCDC.
+
+TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka Consumer reads data, it calculates a new checksum using the same algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka Consumer.
+
+### Checksum V2
+
+Starting from v8.4.0, TiDB and TiCDC introduce Checksum V2 to address issues with Checksum V1 in verifying old values in Update or Delete events after Add Column or Drop Column operations.
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:74]:
```suggestion
Starting from v8.4.0, TiDB and TiCDC introduce Checksum V2 to address issues with Checksum V1 in verifying old values in Update or Delete events after `ADD COLUMN` or `DROP COLUMN` operations.
```

File: ticdc/ticdc-integrity-check.md [hunk 3]
```diff
@@ -67,6 +59,22 @@ TiCDC disables data integrity validation by default. To disable this feature aft
...
 
+## Checksum algorithms
+
+### Checksum V1
+
+Before v8.4.0, TiDB and TiCDC use Checksum V1 for checksum calculation and verification.
+
+After you enable the checksum integrity validation feature for single-row data, TiDB uses the CRC32 algorithm to calculate the checksum of a row and writes it to TiKV along with the data. TiCDC reads the data from TiKV and recalculates the checksum using the same algorithm. If the two checksums are equal, it indicates that the data is consistent during the transmission from TiDB to TiCDC.
+
+TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka Consumer reads data, it calculates a new checksum using the same algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka Consumer.
```
Comment (Oreoxmt) [inline ticdc/ticdc-integrity-check.md:70]:
```suggestion
TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka Consumer reads data, it calculates a new checksum using the same CRC32 algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka Consumer.
```

File: ticdc/ticdc-integrity-check.md [hunk 4]
```diff
@@ -67,6 +59,22 @@ TiCDC disables data integrity validation by default. To disable this feature aft
...
 
+## Checksum algorithms
+
+### Checksum V1
+
+Before v8.4.0, TiDB and TiCDC use Checksum V1 for checksum calculation and verification.
+
+After you enable the checksum integrity validation feature for single-row data, TiDB uses the CRC32 algorithm to calculate the checksum of a row and writes it to TiKV along with the data. TiCDC reads the data from TiKV and recalculates the checksum using the same algorithm. If the two checksums are equal, it indicates that the data is consistent during the transmission from TiDB to TiCDC.
+
+TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka Consumer reads data, it calculates a new checksum using the same CRC32 algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka Consumer.
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:70]:
```suggestion
TiCDC then encodes the data into a specific format and sends it to Kafka. After the Kafka consumer reads the data, it calculates a new checksum using the same CRC32 algorithm as TiDB. If the new checksum is equal to the checksum in the data, it indicates that the data is consistent during the transmission from TiCDC to the Kafka consumer.
```

File: ticdc/ticdc-integrity-check.md [hunk 5]
```diff
@@ -67,6 +59,22 @@ TiCDC disables data integrity validation by default. To disable this feature aft
...
+
+### Checksum V2
+
+Starting from v8.4.0, TiDB and TiCDC introduce Checksum V2 to address issues with Checksum V1 in verifying old values in Update or Delete events after `ADD COLUMN` or `DROP COLUMN` operations.
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:74]:
```suggestion
Starting from v8.4.0, TiDB and TiCDC introduce Checksum V2 to address issues of Checksum V1 in verifying old values in Update or Delete events after `ADD COLUMN` or `DROP COLUMN` operations.
```

File: ticdc/ticdc-integrity-check.md [hunk 6]
```diff
@@ -67,6 +59,22 @@ TiCDC disables data integrity validation by default. To disable this feature aft
...
+
+For new clusters created in v8.4.0 or later, or clusters upgraded to v8.4.0, TiDB uses Checksum V2 by default when single-row data checksum verification is enabled. TiCDC supports handling both Checksum V1 and V2. This change only affects TiDB and TiCDC internal implementation and does not impact checksum calculation methods for downstream Kafka consumers.
```
Comment (qiancai) [inline ticdc/ticdc-integrity-check.md:76]:
```suggestion
For clusters created in v8.4.0 or later, or clusters upgraded to v8.4.0, TiDB uses Checksum V2 by default when single-row data checksum verification is enabled. TiCDC supports handling both Checksum V1 and V2. This change only affects TiDB and TiCDC internal implementation and does not affect checksum calculation methods for downstream Kafka consumers.
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 7]
```diff
@@ -531,4 +531,21 @@ If the message contains the `claimCheckLocation` field, the Kafka consumer reads
 }
 ```
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:533]: These changes are not needed in this PR, right? They are already included in https://github.com/pingcap/docs/pull/18903/files.

**Review by qiancai — APPROVED** [review]

**Pull request merged**